gazebo_ros/src/executor.cpp

```diff
@@ -22,15 +22,22 @@ namespace gazebo_ros
 Executor::Executor()
 : spin_thread_(std::bind(&Executor::run, this))
 {
+  using namespace std::chrono_literals;
   sigint_handle_ = gazebo::event::Events::ConnectSigInt(std::bind(&Executor::shutdown, this));
+  while (!this->spinning) {
+    // TODO(ivanpauno): WARN Terrible hack here!!!!
+    // We cannot call rclcpp::shutdown asynchronously, because it generates exceptions that
+    // cannot be caught properly
```

Comment:
It'd be nice to reference the rclcpp issue here in the comment.

Reply:
See https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1212/commits/018c260a248f04876774189da6491425b4ede6da